hbase-client/src/main/java/org/apache/hadoop/hbase/client/ConnectionImplementation.java

```diff
@@ -1112,6 +1114,11 @@ rpcControllerFactory, getMetaLookupPool(), connectionConfig.getMetaReadRpcTimeou
             ConnectionUtils.getPauseTime(pauseBase, tries), TimeUnit.MILLISECONDS);
         }
       } finally {
+        // update duration of the lock being held
```

Comment:
I missed making this comment before, but can we move the metrics update to after the `unlock`?

Reply:
No, the variable `userRegionLockHeldStartTime` has to be inside/protected by the lock.

Reply:
Okay. I guess we could make a local copy of the variable before leaving the lock, and still update the metric outside of the lock. It's nitpicky, but always good to minimize work under lock.

The fact that we're worried about multithreaded access to this variable also makes me wonder if we can just remove the class variable. `takeUserRegionLock` could return the `userRegionLockHeldStartTime`... or even easier: just record the start time immediately after calling `takeUserRegionLock`. Then we have all the context within `locateRegionInMeta`.

Reply:
Make sense. Let me make the start time of holding the lock as local variable to the function.